### PR TITLE
Feature 41 create answers single select

### PR DESCRIPTION
**Pull request by kaeline**
https://github.com/Metatavu/oss-surveys-api/issues/41
https://github.com/Metatavu/oss-surveys-api/issues/42

### CODE REVIEW

File: src/main/kotlin/fi/metatavu/oss/api/impl/pages/PagesApiImpl.kt [hunk 1]
```diff
@@ -128,23 +130,32 @@ class PagesApiImpl : PagesApi, AbstractApi() {
...
      * @return if is published on any device
      */
-    private suspend fun hasBeenPublished(page: PageEntity): Boolean {
-        if (environment == "staging") return false
-        val belongsToDeviceSurveys = deviceSurveyController.listDeviceSurveysBySurvey(page.survey.id).first
-        return belongsToDeviceSurveys.any {
-            it.status == DeviceSurveyStatus.PUBLISHED ||
-                (it.publishEndTime != null && it.publishEndTime!!.isBefore(OffsetDateTime.now()))
+    private suspend fun canBeModified(page: PageEntity): Response? {
+        if (environment == "staging") return null
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/oss/api/impl/pages/PagesApiImpl.kt:146]: Should this test for both test and staging environments?

File: src/main/kotlin/fi/metatavu/oss/api/impl/pages/answers/SurveyAnswersApiImpl.kt [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    }.asUni()
+
+    @RolesAllowed(UserRole.MANAGER.name)
+    @OptIn(ExperimentalCoroutinesApi::class)
+    override fun findSurveyPageAnswer(
+        surveyId: UUID,
+        pageId: UUID,
+        answerId: UUID
+    ): Uni<Response> = CoroutineScope(vertx.dispatcher()).async {
+        val pageOrError = findSurveyPage(
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/oss/api/impl/pages/answers/SurveyAnswersApiImpl.kt:73]: Shouldn't this be only available in test environment?

File: src/main/kotlin/fi/metatavu/oss/api/impl/pages/answers/SurveyAnswersApiImpl.kt [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    @Inject
+    lateinit var vertx: Vertx
+
+    @RolesAllowed(UserRole.MANAGER.name)
+    @OptIn(ExperimentalCoroutinesApi::class)
+    override fun listSurveyPageAnswers(
+        surveyId: UUID,
+        pageId: UUID
+    ): Uni<Response> = CoroutineScope(vertx.dispatcher()).async {
+        val pageOrError = findSurveyPage(
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/oss/api/impl/pages/answers/SurveyAnswersApiImpl.kt:52]: Shouldn't this be only available in test environment?

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: src/main/kotlin/fi/metatavu/oss/api/impl/pages/answers/SurveyAnswersApiImpl.kt (hunk 2 above)
Comment (kaeline) [reply src/main/kotlin/fi/metatavu/oss/api/impl/pages/answers/SurveyAnswersApiImpl.kt:69]: Don't we want managers to see answers for the page in normal env as well?

File: src/main/kotlin/fi/metatavu/oss/api/impl/devicesurveys/devicesurveydata/DeviceSurveyDataApiImpl.kt [hunk 4]
```diff
@@ -65,4 +79,49 @@ class DeviceSurveyDataApiImpl: fi.metatavu.oss.api.spec.DeviceDataApi, AbstractA
...
+            "No question found for page $pageId"
+        )
+
+        val deviceSurvey = deviceSurveyController.findDeviceSurvey(deviceSurveyId)
+            ?: return@async createNotFoundWithMessage(
+                target = DEVICE_SURVEY,
+                id = deviceSurveyId
+            )
+
+        if (deviceSurvey.device != device) {
```
[the remark this answers is not in the document]
Comment (kaeline) [reply src/main/kotlin/fi/metatavu/oss/api/impl/devicesurveys/devicesurveydata/DeviceSurveyDataApiImpl.kt]: true, fixed

File: src/test/kotlin/fi/metatavu/oss/api/test/functional/resources/LocalTestProfile.kt [hunk 5]
```diff
@@ -12,7 +12,8 @@ class LocalTestProfile: QuarkusTestProfile {
             "quarkus.scheduler.enabled" to "true",
             "scheduled.survey.publish.interval" to "5s",
             "environment" to "test",
-            "mqtt.base.topic" to "test"
+            "mqtt.base.topic" to "test",
+            "environment" to "staging"
```
[the remark this answers is not in the document]
Comment (kaeline) [reply src/test/kotlin/fi/metatavu/oss/api/test/functional/resources/LocalTestProfile.kt]: done

**Review by anttileppa — APPROVED** [review]

**Pull request merged**